lib/libzfs/libzfs_dataset.c

```diff
@@ -196,6 +196,16 @@ zfs_validate_name(libzfs_handle_t *hdl, const char *path, int type,
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "reserved disk name"));
 				break;
+			
+			case NAME_ERR_SELF_REFERENCE:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "self reference is found"));
```

Comment:
I think it might be easier for admins to understand a more specific error message, like `"dot ('.') component in name"`.  You could check the error message in context to see how it flows, e.g.:
```
$ zfs create test/.
bla bla bla
self reference is found
```

Reply:
how about? self reference, ('.') is found in name